boa/cli/mambabuild.py

```diff
@@ -19,6 +20,15 @@
 solver_map = {}
 
 
+def _get_solver(subdir):
```

Comment:
great! This function is a little confusign because it relies on the fact that `subdir` is passed by reference, and that Python implicitly returns None.

I think it would be better to actually instantiate the solver inside this function, add it to the map and return it from this function. So that this function always returns a solver.

Or we could make a function "normalize_subdir" that returns the correct subdir for the solver, and instantiate the solver in the other function.
But I think there are some advanced, somewhat surprising Python features in play here.

Reply:
@wolfv I see what you're suggesting and it makes sense. I was going to do initially like you suggested but I wasn't sure if passing the `channels_urls` and `output_dir` to the `_get_solver()` function was the best choice.
I also extracted the `normalize_subdir` into `boa.core.utils` module and used both in `mambabuild.py` and `solver.py`.